client/src/components/QualityManagement/Parts.jsx

```diff
@@ -0,0 +1,328 @@
+import { DeleteIcon, SmallAddIcon } from '@chakra-ui/icons';
+
+const Parts = () => {
+  const { isLoading, isSuccess, data, refetch } = useQuery('defects', getDefects);
+  const [page, setPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+  const [selected, setSelected] = useState([]);
+
+  // Modal (add new defect)
+  const [partName, setPartName] = useState('');
+  const type = useRef('');
+  const status = useRef('');
+  const description = useRef('');
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [isLoadingButton, setIsLoadingButton] = useState(false);
+  const toast = useToast();
+
+  const handleChangePage = (e, newPage) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (e) => {
+    setRowsPerPage(parseInt(e.target.value, 10));
+    setPage(0);
+  };
+
+  const handleStatusColor = (status) => {
+    switch (status) {
+      case 'Solved':
+        return 'green';
+      case 'Pending':
+        return 'orange';
+      case 'Ongoing':
+        return 'yellow';
+    }
+  };
+
+  const handlePartNameInput = (e) => {
+    setPartName(e.target.value);
+  };
+
+  const isSelected = (_id) => selected.indexOf(_id) !== -1;
+
+  const handleSelectAllClick = (e) => {
+    if (e.target.checked) {
+      const newSelecteds = data.data.map((n) => n._id);
+      setSelected(newSelecteds);
+      return;
+    }
+    setSelected([]);
+  };
+
+  const handleOnChange = (e, _id) => {
+    const selectedIndex = selected.indexOf(_id);
+    let newSelected = [];
+
+    if (selectedIndex === -1) {
+      newSelected = newSelected.concat(selected, _id);
+    } else if (selectedIndex === 0) {
+      newSelected = newSelected.concat(selected.slice(1));
+    } else if (selectedIndex === selected.length - 1) {
+      newSelected = newSelected.concat(selected.slice(0, -1));
+    } else if (selectedIndex > 0) {
+      newSelected = newSelected.concat(
+        selected.slice(0, selectedIndex),
+        selected.slice(selectedIndex + 1)
+      );
+    }
+
+    setSelected(newSelected);
+  };
+
+  const handleDelete = async (e) => {
+    e.preventDefault();
+    try {
+      await deleteDefects(selected);
+      toast({
+        title: 'Deleted',
+        description: 'Defects have been deleted successfully',
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+      });
+      setSelected([]);
+      refetch();
+    } catch (err) {
+      toast({
+        title: 'Error',
+        description: err.message,
+        status: 'error',
+        duration: 9000,
+        isClosable: true,
+      });
+    }
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setIsLoadingButton(true);
+    try {
+      await createDefect({
+        partName: partName,
+        type: type.current.value,
+        status: status.current.value,
+        description: description.current.value,
+        onCloseComplete: onClose,
```

Comment:
Does `onCloseComplete` belong in the post request body?

Reply:
LOL good catch! it definitely doesn't belong there